crates/ethereum-forks/src/hardforks/mod.rs

```diff
@@ -0,0 +1,128 @@
+/// Ethereum helper methods
+mod ethereum;
+pub use ethereum::EthereumHardforks;
+
+/// Optimism helper methods
+mod optimism;
+pub use optimism::OptimismHardforks;
+
+use crate::{ForkCondition, Hardfork};
+use std::collections::HashMap;
+
+/// Generic trait over a set of ordered hardforks
+pub trait Hardforks: Default + Clone {
+    /// Retrieves [`ForkCondition`] from `fork`. If `fork` is not present, returns
+    /// [`ForkCondition::Never`].
+    fn fork<H: Hardfork>(&self, fork: H) -> ForkCondition;
+
+    /// Get an iterator of all hardforks with their respective activation conditions.
+    fn forks_iter(&self) -> impl Iterator<Item = (&Box<dyn Hardfork + 'static>, ForkCondition)>;
+
+    /// Convenience method to check if a fork is active at a given timestamp.
+    fn is_fork_active_at_timestamp<H: Hardfork>(&self, fork: H, timestamp: u64) -> bool {
+        self.fork(fork).active_at_timestamp(timestamp)
+    }
+
+    /// Convenience method to check if a fork is active at a given block number.
+    fn is_fork_active_at_block<H: Hardfork>(&self, fork: H, block_number: u64) -> bool {
+        self.fork(fork).active_at_block(block_number)
+    }
+}
+
+/// Ordered list of a chain hardforks that implement [`Hardfork`].
+#[derive(Default, Clone, PartialEq, Eq)]
+pub struct ChainHardforks {
+    forks: Vec<(Box<dyn Hardfork>, ForkCondition)>,
+    map: HashMap<&'static str, ForkCondition>,
```

Comment:
we need to lookup by name or `Hardfork` and also need a ordered list, 
I assume looking up in a map would be better instead of iterating and comparing? but maybe not?

Reply:
Sure